test/parallel/test-eventtarget.js

```diff
@@ -176,14 +176,16 @@ let asyncTest = Promise.resolve();
 }
 
 {
-  const uncaughtException = common.mustCall((err, event) => {
+  const uncaughtException = common.mustCall((err, origin) => {
     strictEqual(err.message, 'boom');
-    strictEqual(event.type, 'foo');
+    strictEqual(origin, 'uncaughtException');
   }, 4);
 
-  // Whether or not the handler function is async or not, errors
-  // are routed to uncaughtException
-  process.on('error', uncaughtException);
+  // Make sure that we no longer call 'error' on error.
+  process.on('error', common.mustNotCall());
+  // Don't call rejection even for async handlers.
+  process.on('unhandledRejection', common.mustNotCall());
```

Comment:
I suspect we need (in a future PR) something like `captureRejections` from EventEmitter if there is no `error` between the async function throwing and `uncaughtException`.

Reply:
( and then possibly to remove the `async` behaviour entirely)